query-connector/src/app/database-service.ts

```diff
@@ -117,43 +101,18 @@ export const executeDefaultQueryCreation = async () => {
  */
 export const mapQueryRowsToValueSets = async (rows: QueryResultRow[]) => {
   // Create groupings of rows (each of which is a single Concept) by their ValueSet ID
-  const vsIdGroupedRows = rows.reduce((conceptsByVSId, r) => {
-    if (!(r["valueset_id"] in conceptsByVSId)) {
-      conceptsByVSId[r["valueset_id"]] = [];
-    }
-    conceptsByVSId[r["valueset_id"]].push(r);
-    return conceptsByVSId;
-  }, {});
-
-  // Each "prop" of the struct is now a ValueSet ID
-  // Iterate over them to create formal Concept Groups attached to a formal VS
-  const valueSets = Object.keys(vsIdGroupedRows).map((vsID) => {
-    const conceptGroup: QueryResultRow[] = vsIdGroupedRows[vsID];
-    const valueSet: ValueSet = {
-      valueSetId: conceptGroup[0]["valueset_id"],
-      valueSetVersion: conceptGroup[0]["version"],
-      valueSetName: conceptGroup[0]["valueset_name"],
-      // External ID might not be defined for user-defined valuesets
-      valueSetExternalId: conceptGroup[0]["valueset_external_id"]
-        ? conceptGroup[0]["valueset_external_id"]
-        : undefined,
-      author: conceptGroup[0]["author"],
-      system: conceptGroup[0]["code_system"],
-      ersdConceptType: conceptGroup[0]["type"]
-        ? conceptGroup[0]["type"]
-        : undefined,
-      dibbsConceptType: conceptGroup[0]["dibbs_concept_type"],
-      includeValueSet: conceptGroup.find((c) => c["include"]) ? true : false,
-      concepts: conceptGroup.map((c) => {
-        return {
-          code: c["code"],
-          display: c["display"],
-          include: c["include"],
-        };
-      }),
-    };
-    return valueSet;
-  });
+  const valueSets: ValueSet[] = rows
+    .map((curRow) => {
```

Comment:
I think the key functional difference is that the new column has an extra layer of nesting (condition) that before we normalized away the condition <> valueset join. This was to solve the edge case of a single valueset being in multiple conditions / not having a way to backassociate valueset --> condition when we just had valueset information. The change sticks that in natively to the query data column. This is why we have the extra level of `map` and `flat`.

Let me know if you're finding the original function more readable because 1) the fields are enumerated / set accordingly or 2) because it has one fewer dimension of mapping or 3) for some other reason. Happy to change it back if 1), but I might need more info for what it should look like if it's 2) or 3)

Reply:
Ah, even though we had talked about it yesterday, I had forgotten the extra layer of nesting with condition.

I think the original function is more readable because of 1.

Reply:
Went ahead and broke this out! hopefully this covers what you were suggesting